lightning/src/routing/network_graph.rs

```diff
@@ -1658,4 +1747,132 @@ mod tests {
 		network.write(&mut w).unwrap();
 		assert!(<NetworkGraph>::read(&mut ::std::io::Cursor::new(&w.0)).unwrap() == *network);
 	}
+
+	#[test]
+	fn network_graph_score_payment() {
+		// Set up a network that consists of just a channel between node_1 and node_2
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_1_privkey = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_2_privkey = &SecretKey::from_slice(&[41; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_1_privkey);
+		let node_id_2 = PublicKey::from_secret_key(&secp_ctx, node_2_privkey);
+		let node_1_btckey = &SecretKey::from_slice(&[40; 32]).unwrap();
+		let node_2_btckey = &SecretKey::from_slice(&[39; 32]).unwrap();
+
+		let short_channel_id = 0;
+		let chain_hash = genesis_block(Network::Testnet).header.bitcoin_hash();
+
+		{
+			// There is no nodes in the table at the beginning.
+			let network = net_graph_msg_handler.network_graph.read().unwrap();
+			assert_eq!(network.get_nodes().len(), 0);
+		}
+
+		{
+			// Announce a channel we will update
+			let unsigned_announcement = UnsignedChannelAnnouncement {
+				features: ChannelFeatures::empty(),
+				chain_hash,
+				short_channel_id,
+				node_id_1,
+				node_id_2,
+				bitcoin_key_1: PublicKey::from_secret_key(&secp_ctx, node_1_btckey),
+				bitcoin_key_2: PublicKey::from_secret_key(&secp_ctx, node_2_btckey),
+				excess_data: Vec::new(),
+			};
+
+			let msghash = hash_to_message!(&Sha256dHash::hash(&unsigned_announcement.encode()[..])[..]);
+			let valid_channel_announcement = ChannelAnnouncement {
+				node_signature_1: secp_ctx.sign(&msghash, node_1_privkey),
+				node_signature_2: secp_ctx.sign(&msghash, node_2_privkey),
+				bitcoin_signature_1: secp_ctx.sign(&msghash, node_1_btckey),
+				bitcoin_signature_2: secp_ctx.sign(&msghash, node_2_btckey),
+				contents: unsigned_announcement.clone(),
+			};
+			match net_graph_msg_handler.handle_channel_announcement(&valid_channel_announcement) {
+				Ok(_) => (),
+				Err(_) => panic!()
+			};
+		}
+
+		{
+			// At the start, the node_1 <-> node_2 channel's DirectionalChannelInfos should be 0.
+			let mut network = net_graph_msg_handler.network_graph.write().unwrap();
+			let chan_id = network.get_nodes().get(&node_id_1).unwrap().channels[0];
+			let channel_info = network.channels.get_mut(&chan_id).unwrap();
+			// Assign a DirectionalChannelInfo and ChannelScore object to one_to_two of
```

Comment:
`one_to_two` is a awful name, if you have a better idea for naming you got my Concept ACK (`alice_to_bob` ?)

Reply:
I like `alice_to_bob`. Going to go with that

Reply:
Alice and Bob are suitable in test code when referring to specific examples. However, I don't think the convention should be extended to production code when referring to a general node.